src/Common/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useState } from 'react'
+import { Page, PageValueItemProps, PaginationProps } from './types'
+import { createPageArr } from './utils'
+import './pagination.scss'
+import PageSizeSelector from './PageSizeSelector'
+
+const PageValueItem = ({ value, isSelected, selectPage }: PageValueItemProps) => {
+    const classes = isSelected ? 'page__button page__button--selected' : 'page__button'
+    const handlePageSelection = () => {
+        selectPage(value)
+    }
+
+    return (
+        <li className="page">
+            <button className={classes} onClick={handlePageSelection} type="button">
+                {value}
+            </button>
+        </li>
+    )
+}
+
+// NOTE: This component will heavily benefit from data normalization
+const Pagination = ({
+    size,
+    pageSize,
+    offset,
+    changePage,
+    changePageSize,
+    isPageSizeFix,
+    pageSizeOptions,
+    rootClassName,
+}: PaginationProps) => {
+    const [pages, setPages] = useState<Page[]>([])
+
+    useEffect(() => {
+        const _pages = createPageArr({ size, pageSize, selectedPageNo: 1 })
+        setPages(_pages)
+    }, [size, pageSize])
+
+    useEffect(() => {
+        const selectedPageNo = 1 + Math.floor(offset / pageSize)
+        const _pages = createPageArr({ size, pageSize, selectedPageNo })
+        setPages(_pages)
```

Comment:
We can create a common fn _updatePages that calls the createPageArr along with setPages so that it is following DRY

Reply:
imported the logic from dashboard, even have left a note stating normalisation might benefit here.